cli/cli.ts

```diff
@@ -3500,6 +3500,7 @@ function testForBuildTargetAsync(useNative: boolean, cachedSHA: string): Promise
 
     return mainPkg.loadAsync()
         .then(() => {
+            mainPkg.ignoreTests = true
```

Comment:
@riknoll , this function is called "testForBuildTargetAsync" which makes me think it is just checking some state but it in fact is triggering several compilations and the outputs of those are going into the API cache.

Can you recommend a better name for this? Seems more like: checkAndUpdateApiCacheForPackage where "mainPkg" is an implicit argument (which maybe should be explicit?)